constructor pragma leads to "Most Vexing Parse" in c++ code gen

```nim
type StdString {.importCpp: "std::string", header: "<string>", byref.} = object
proc initString(): StdString {.constructor, importCpp: "std::string(@)", header: "<string>".}
proc size(this: var StdString): csize {.importCpp: "size", header: "<string>".}

proc f(): csize  =
  var myString: StdString = initString()
  return myString.size()

echo f()
```
generates

```cpp
N_NIMCALL(size_t, f_yTiPEX7pvNl083AJkerm5g)(void) {
        size_t result;
        nimfr_("f", "test.nim");
{       result = (size_t)0;
        nimln_(6, "test.nim");
        std::string myString();   // <<<< this line
        nimln_(7, "test.nim");
        result = myString.size();
        goto BeforeRet_;
        }BeforeRet_: ;
        popFrame();
        return result;
}
```

Unfortunately, due to the "[Most Vexing Parse](https://en.wikipedia.org/wiki/Most_vexing_parse)" that line is parsed as declaring `myString` to be a nullary function that returns a std::string, rather than a default-constructed std::string value. This causes a very confusing error message:

```
> nim cpp  -r test.nim
Hint: used config file '/home/mstearn/.choosenim/toolchains/nim-0.17.2/config/nim.cfg' [Conf]
Hint: used config file '/home/mstearn/nim/tmp/nim.cfg' [Conf]
Hint: system [Processing]
Hint: test [Processing]
CC: test
CC: stdlib_system
CC: stdlib_sharedlist
CC: stdlib_locks
Error: execution of an external compiler program 'g++ -c  -w -w -fpermissive -pthread  -I/home/mstearn/.choosenim/toolchains/nim-0.17.2/lib -o /home/mstearn/nim/tmp/nimcache/test.o /home/mstearn/nim/tmp/nimcache/test.cpp' failed with exit code: 1

/home/mstearn/nim/tmp/nimcache/test.cpp: In function ‘size_t f_yTiPEX7pvNl083AJkerm5g()’:
/home/mstearn/nim/tmp/nimcache/test.cpp:85:20: error: request for member ‘size’ in ‘myString’, which is of non-class type ‘std::__cxx11::string() {aka std::__cxx11::basic_string<char>()}’
  result = myString.size();
                    ^~~~
```

Removing the `constructor` pragma from the initString() declaration makes this problem go away, but the documentation implies that that only works for copyable/movable types (until c++17 at least).

--- a/compiler/ccgstmts.nim
+++ b/compiler/ccgstmts.nim
@@ -243,7 +243,10 @@ proc genSingleVar(p: BProc, a: PNode) =
           if params != nil: params.add(~", ")
           assert(sonsLen(typ) == sonsLen(typ.n))
           add(params, genOtherArg(p, value, i, typ))
-        lineF(p, cpsStmts, "$#($#);$n", [decl, params])
+        if params == nil:
+          lineF(p, cpsStmts, "$#;$n", [decl])
+        else:
+          lineF(p, cpsStmts, "$#($#);$n", [decl, params])
       else:
         initLocExprSingleUse(p, value, tmp)
         lineF(p, cpsStmts, "$# = $#;$n", [decl, tmp.rdLoc])

--- a/tests/cpp/temitlist.nim
+++ b/tests/cpp/temitlist.nim
@@ -1,6 +1,7 @@
 discard """
   targets: "cpp"
-  output: '''6.0'''
+  output: '''6.0
+0'''
 """
 
 # bug #4730
@@ -20,3 +21,16 @@ proc main =
   echo v[0]
 
 main()
+
+#------------
+
+#bug #6837
+type StdString {.importCpp: "std::string", header: "<string>", byref.} = object
+proc initString(): StdString {.constructor, importCpp: "std::string(@)", header: "<string>".}
+proc size(this: var StdString): csize {.importCpp: "size", header: "<string>".}
+
+proc f(): csize =
+  var myString: StdString = initString()
+  return myString.size()
+
+echo f()

